compiler/loco/include/loco/IR/DataTypeTraits.h

```diff
@@ -123,6 +129,8 @@ inline uint32_t size(DataType data_type)
 {
   switch (data_type)
   {
+    case DataType::S4:
+      return sizeof(DataTypeImpl<DataType::S4>::Type);
```

Comment:
`loco::size` is used to estimate the size of an input tensor or a scratchpad tensor.

```
$ git grep loco::size

compiler/circle-eval-diff/src/InputDataLoader.cpp:    const auto dtype_size = loco::size(input_node->dtype());
compiler/circle-eval-diff/src/InputDataLoader.cpp:    size_t byte_size = loco::size(input_node->dtype());
compiler/circle-execution-plan/pal/ScratchpadHelperLinux.h:    scratchpad_sizes.push_back(scratchpad_size * loco::size(lhs->dtype()));
compiler/circle-execution-plan/pal/ScratchpadHelperLinux.h:    scratchpad_sizes.push_back(scratchpad_size * loco::size(rhs->dtype()));
compiler/circle-interpreter/src/CircleInterpreter.cpp:  uint32_t tensor_size = loco::size(node->dtype());
compiler/circle-mpqsolver/src/core/ErrorMetric.cpp:      size_t cur_size = first_elementary.size() / loco::size(loco::DataType::FLOAT32);
compiler/circle-mpqsolver/src/core/Evaluator.cpp:  uint32_t tensor_size = loco::size(node->dtype());
compiler/circle-part-driver/src/PModelsRunner.cpp:  uint32_t tsize = loco::size(node->dtype());
compiler/common-artifacts/src/TestDataGenerator.cpp:      auto dtype_size = loco::size(input_node->dtype());
compiler/common-artifacts/src/TestDataGenerator.cpp:      uint32_t tensor_bytesize = loco::size(output_node->dtype());
compiler/dalgona/src/Dalgona.cpp:  uint32_t dtype_size = loco::size(node->dtype());
compiler/embedded-import-value-test/src/TestDriver.cpp:  uint32_t tensor_size = loco::size(node->dtype());
compiler/embedded-import-value-test/src/TestDriver.cpp:  assert(inputs_data.size() % loco::size(dtype) == 0); // FIX ME UNLESS
compiler/embedded-import-value-test/src/TestDriver.cpp:  const auto element_count = inputs_data.size() / loco::size(dtype);
compiler/luci-eval-driver/src/EvalDriver.cpp:  uint32_t tensor_size = loco::size(node->dtype());
compiler/luci-interpreter/include/luci_interpreter/core/DataType.h:inline size_t getDataTypeSize(DataType data_type) { return loco::size(data_type); }
compiler/record-minmax/src/RecordMinMax.cpp:  uint32_t tensor_size = loco::size(node->dtype());
```

If loco::size(S4) returns 1, there may be a problem in estimating the size of a tensor. Could you throw an exception for `DataType::S4`? I guess there will be no problem for now.

```suggestion
      throw std::runtime_error("S4 type is not supported by loco::size");
```

Reply:
With throw, common-artifacts fails to generate data

> terminate called after throwing an instance of 'std::runtime_error'
  what():  S4 type is not supported by loco::size
Aborted (core dumped)
gmake[3]: *** [compiler/common-artifacts/CMakeFiles/common_artifacts_deps.dir/build.make:7648: compiler/common-artifacts/FullyConnected_I4_001.opt/metadata/tc/input.h5] Error 134

Reply:
um... maybe introduce `luci::size()` as `luci` will store as 1 byte.

Reply:
> FullyConnected_I4_001

This may be resolved by https://github.com/Samsung/ONE/pull/12743/files#r1525742386

>  introduce luci::size() as luci will store as 1 byte.

+1 for this.